packages/next/src/build/webpack/loaders/next-app-loader.ts

```diff
@@ -51,6 +70,62 @@ async function createTreeCodeFromPath({
   let rootLayout: string | undefined
   let globalError: string | undefined
 
+  async function discoverStaticMetadataFiles(
+    routerDirPath: string,
+    isDev: boolean
+  ) {
+    let hasStaticMetadataFiles = false
+    const iconsMetadata: {
+      icon: string[]
+      apple: string[]
+    } = {
+      icon: [],
+      apple: [],
+    }
+
+    // collect metadata if there's any from the folder of the page
+    const resolvedRouteDir = await resolve(routerDirPath, true)
+    const isDirectory = resolvedRouteDir
+      ? (await fs.stat(resolvedRouteDir)).isDirectory()
+      : false
+
+    if (resolvedRouteDir && isDirectory) {
+      const files = await fs.readdir(resolvedRouteDir, {
+        encoding: 'utf-8',
+        withFileTypes: true,
+      })
```

Comment:
Note: This is not watch the directory correctly.
e. g. adding files will not invalidate this module.

Hint: use `this.addMissingDependency` for all files that might appear.

Reply:
changed to limited compositions